src/components/LoanCards/SharedLoanCardController.vue

```diff
@@ -0,0 +1,307 @@
+import { setLendAmount, handleInvalidBasket, hasBasketExpired } from '@/util/basketUtils';
+import { readLoanFragment, watchLoanData } from '@/util/loanUtils';
+import bookmarkLoan from '@/util/bookmarkUtil';
+import logFormatter from '@/util/logFormatter';
+import { createIntersectionObserver } from '@/util/observerUtils';
+import percentRaisedMixin from '@/plugins/loan/percent-raised-mixin';
+import timeLeftMixin from '@/plugins/loan/time-left-mixin';
+import loanCardFieldsExtendedFragment from '@/graphql/fragments/loanCardFieldsExtended.graphql';
+import KvClassicLoanCard from '~/@kiva/kv-components/vue/KvClassicLoanCard';
+
+const PHOTO_PATH = 'https://www-kiva-org.freetls.fastly.net/img/';
+
+const loanQuery = gql`
+	${loanCardFieldsExtendedFragment}
+	query kcBasicLoanCard($basketId: String, $loanId: Int!) {
+	shop (basketId: $basketId) {
+		id
+		basket {
+			id
+			# for isInBasket
+			items {
+				values {
+					id
+				}
+			}
+		}
+	}
+	lend {
+		loan(id: $loanId) {
+			id
+			...loanCardFieldsExtended
+		}
+	}
+	my {
+		id
+		userAccount {
+			id
+		}
+	}
+}`;
+
+export default {
+	name: 'SharedLoanCardController',
+	props: {
+		loanId: {
+			type: Number,
+			required: true,
+		},
+		customLoanDetails: {
+			type: Boolean,
+			default: false
+		},
+		useFullWidth: {
+			type: Boolean,
+			default: false
+		},
+		showTags: {
+			type: Boolean,
+			default: false
+		},
+		categoryPageName: {
+			type: String,
+			default: '',
+		},
+		enableFiveDollarsNotes: {
+			type: Boolean,
+			default: false
+		},
+		enableMoreCta: {
+			type: Boolean,
+			default: false
+		},
+		largeCard: {
+			type: Boolean,
+			default: false
+		},
+		userBalance: {
+			type: String,
+			default: undefined
+		},
+		showViewLoan: {
+			type: Boolean,
+			default: false
+		},
+	},
+	inject: ['apollo', 'cookieStore'],
+	mixins: [percentRaisedMixin, timeLeftMixin],
+	components: {
+		KvClassicLoanCard
+	},
+	data() {
+		return {
+			loan: null,
+			basketItems: null,
+			isLoading: true,
+			queryObserver: null,
+			viewportObserver: null,
+			isAdding: false,
+			loanCallouts: undefined,
+			isVisitor: true,
+			isBookmarked: false,
+			currentRoute: {},
+			PHOTO_PATH,
+		};
+	},
+	methods: {
+		createViewportObserver() {
+			// Watch for this element being in the viewport
+			this.viewportObserver = createIntersectionObserver({
+				targets: [this.$el],
+				callback: entries => {
+					entries.forEach(entry => {
+						if (entry.target === this.$el && entry.intersectionRatio > 0) {
+							// This element is in the viewport, so load the data.
+							this.loadData();
+						}
+					});
+				}
+			});
+			if (!this.viewportObserver) {
+				// Observer was not created, so call loadData right away as a fallback.
+				this.loadData();
+			}
+		},
+		destroyViewportObserver() {
+			if (this.viewportObserver) {
+				this.viewportObserver.disconnect();
+			}
+		},
+		loadData() {
+			if (!this.queryObserver) {
+				this.queryObserver = watchLoanData({
```

Comment:
I know we don't do this in the old `ui` card that this is based off of, but make sure this subscription is unsubscribed in `beforeDestroy`. See the `onBeforeUnmount` in the CPS component for an example.